app/assets/stylesheets/application.scss

```diff
@@ -12,10 +12,12 @@
  *
  */
 @import "bootstrap-sprockets";
+@import "bootstrap";
```

Comment:
Yep that can definitely be confusing.  Sometimes you want to override bootstrap variables.  Other times you want to override their styling.  One has to go before `import bootstrap`, the other has to after it.

The reason I created a file specifically for overriding variables is because of the position where it needs to be imported.  Anything overriding bootstrap styling can go in any other asset file in app/assets/stylesheets/*.scss because those will **all** be imported after bootstrap, so it really doesn't matter what file you put it in with regards to its connection to bootstrap.